src/torchmetrics/functional/regression/minkowski.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+from torchmetrics.utilities.exceptions import TorchMetricsUserError
+
+
+def _minkowski_distance_update(preds: Tensor, targets: Tensor, p: float) -> Tensor:
+    """Updates and returns variables required to compute Minkowski distance.
+
+    Checks for same shape of input tensors.
+
+    Args:
+        preds: Predicted tensor
+        target: Ground truth tensor
+        p: Non-negative number acting as the exponent to the errors
+    """
+    _check_same_shape(preds, targets)
+
+    if p < 0:
```

Comment:
lets make the check more strict
```suggestion
    if not isinstance(p, float) and p < 0:
```